modules/integration/src/test/scala/scala/cli/integration/FmtTests.scala

```diff
@@ -57,4 +72,28 @@ class FmtTests extends munit.FunSuite {
     }
   }
 
+  test("filter correctly with --check") {
+    simpleInputsWithFilter.fromRoot { root =>
+      val out      = os.proc(TestUtil.cli, "fmt", ".", "--check").call(cwd = root).out.text.trim
```

Comment:
Did you check, if scala-cli overrides defined `runner.dialect = scala213` in `.scalafmt.conf` by cli options?

Scala CLi resolve `runner.dialect` base on scala params [here](https://github.com/VirtusLab/scala-cli/blob/b754d2afdda114e97febfb0090773cc582bafd19/modules/cli/src/main/scala/scala/cli/commands/Fmt.scala#L160) and then pass to scalafmt args

Reply:
```
▶ cat .scalafmt.conf
version = 3.5.2
runner.dialect = scala3
project {
  includePaths = [
    "glob:**.scala",
    "regex:.*\\.sc"
  ]
  excludePaths = [
    "glob:**/src/test/scala/**.scala"
  ]
}
 
▶ scala-cli-prototype fmt . --dialect scala213
/Users/pchabelski/IdeaProjects/scala-cli-tests/scalafmt-dialects/s3.scala:4: error: [dialect scala3] identifier expected but class found
enum class Weekday {
     ^
 ```
 seems to work correctly. whatever is defined in `.scalafmt.conf` has priority, as it should.
 if necessary, I can add an integration test for this, as I don't see one. Thoughts?